libcontainer/factory_linux.go

```diff
@@ -261,19 +265,26 @@ func (l *LinuxFactory) StartInitialization() (err error) {
 			fmt.Fprintln(os.Stderr, err)
 			return
 		}
-		// ensure that this pipe is always closed
-		pipe.Close()
 	}()
 	defer func() {
 		if e := recover(); e != nil {
 			err = fmt.Errorf("panic from initialization: %v, %v", e, string(debug.Stack()))
 		}
 	}()
+
+	var i initer
 	i, err = newContainerInit(it, pipe, rootfd)
 	if err != nil {
 		return err
 	}
-	return i.Init()
+
+	if i, err = newContainerInit(it, pipe, rootfd); err != nil {
+		return
```

Comment:
You can't use ```i, err :=``` as there are no new variables so it would fail compile.

Is there a reason for not using straight ```return``` as this is not only the idiomatic way, given the use of ```err``` in the ```defer``` not doing so could easily lead to additional coding errors where the defer fails to get the required error.

Given this I would argue that its both clearer and less error prone to use the short ```return``` format.